workspace/workspace.go

```diff
@@ -198,9 +198,21 @@ func (ws Workspace) SolutionDir(s string) (string, error) {
 		if _, err := os.Lstat(path); os.IsNotExist(err) {
 			return "", err
 		}
-		if _, err := os.Lstat(filepath.Join(path, solutionFilename)); err == nil {
+		if err := checkSolutionFile(path); err == nil {
 			return path, nil
 		}
 		path = filepath.Dir(path)
 	}
 }
+
+func checkSolutionFile(path string) error {
+	legacySolutionPath := filepath.Join(path, ".solution.json")
+	solutionPath := filepath.Join(path, SolutionMetadataFilepath())
+
+	if _, err := os.Lstat(legacySolutionPath); err == nil {
+		return migrateLegacySolutionFile(legacySolutionPath, solutionPath)
```

Comment:
If we don’t rename and instead warn the use that a deprecated solution files exists would it make the migration easier. @kytrinyx @nobbz thoughts on leaving the file or renaming it for the user. 

I recommended renaming but I’m not sure how users will feel about the CLI renaming